src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-PostgreSQL.json

```diff
@@ -74,8 +74,16 @@
     },
     "policyRule": {
       "if": {
-        "field": "type",
-        "equals": "Microsoft.DBforPostgreSQL/servers"
+        "anyOf": [
+          {
+            "field": "type",
+            "equals": "Microsoft.DBforPostgreSQL/flexibleServers"
+          },
+          {
+            "field": "type",
+            "equals": "Microsoft.DBforPostgreSQL/servers"
+          }
```

Comment:
Not sure how I feel about this. Looping in @matt-FFFFFF for his inputs also.

1. This is very neat and clever update to this policy to handle multiple resource types 💪
    - We did similar recently in #1216 to handle different app service tiers/skus as they have different categories
2. However, I'm concerned that whilst this is technically for the same RP namespace, they are for different resources.
    - So therefore does this better live split out into a new custom policy definition and added to the existing initiative? 

**Splitting it out**, keeps it simple to understand and update for other consumers and not have to see or work out the clever logic going on

**Keeping it as it is**, it's a really smart approach and reduces the amount of custom policies in total we have to maintain.

Im leaning towards **splitting it out**, but I've not got strong opinions either way (but am happy to make a call if needs be). This is why I've looped in my colleague @matt-FFFFFF to give his opinion on the matter to see if he has a stronger preference either way and any other reasoning as to why.

@matt-FFFFFF let us know your thoughts when you get 5

Reply:
Thanks for looping me in @jtracey93 

I think on balance here I am happy with this in the same policy definition.

My thinking is that these are both Postgres resources and there probably won't be very many different types in the future. Hence, using resource conditions is a reasonable approach as it will remain maintainable.

This does also make it slightly simpler to manage the overall policy set for diagnostic settings as well.

Reply:
Splitting them up is feasible, though overall its more code to manage.

Reply:
Cool thanks @matt-FFFFFF for your input here. 

With that in mind let's leave it as is. I'll mark this as resolved @chrholt 

Just the other comments to resolve 👍